File: app/src/main/java/tech/thdev/app/view/main/presenter/MainPresenter.java
package tech.thdev.app.view.main.presenter;

import tech.thdev.app.adapter.pager.model.SectionsPagerModel;
import tech.thdev.base.presenter.AbstractPresenter;

/**
 * Created by tae-hwan on 12/11/16.
 */

public class MainPresenter extends AbstractPresenter<MainContract.View> implements MainContract.Presenter {

    private SectionsPagerModel sectionsPagerModel;

    @Override
    public void setSectionPagerModel(SectionsPagerModel sectionPagerModel) {
        this.sectionsPagerModel = sectionPagerModel;
    }

    @Override
    public void loadSectionPagerItem() {
        sectionsPagerModel.setListItem(1);
        sectionsPagerModel.setListItem(2);
        sectionsPagerModel.setListItem(3);

        getView().updatePager();
    }
}

File: app/src/main/java/tech/thdev/app/view/presenter/MainPresenter.kt
package tech.thdev.app.view.presenter

import android.content.Context
import android.os.Build
import android.provider.Settings

/**
 * Created by tae-hwan on 18/12/2016.
 */
class MainPresenter : MainContract.Presenter {

    override lateinit var view: MainContract.View

    override fun startOverlayWindowService(context: Context?) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
            && !Settings.canDrawOverlays(context)
        ) {
            view.showObtainingPermissionOverlayWindow()
        } else {
            view.showStartOverlay()
        }
    }

    override fun onOverlayResult(context: Context?) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            if (Settings.canDrawOverlays(context)) {
                view.showStartOverlay()
            }
        }
    }
}